src/interface.jl

```diff
@@ -1,13 +1,13 @@
 
-struct DP5Iterator{T <: Real}
+struct SolverIterator{T <: Real}
     solver::AbstractDPSolver
     times::AbstractVector{T}
 end
 
 # gets the first (t,y), return index which is the state
 # here we choose 2 because 1 is the initial state which 
 # is has been returned by the iterator
-function Base.iterate(dp5_iterator::DP5Iterator)
+function Base.iterate(dp5_iterator::SolverIterator)
```

Comment:
I think it best to rename the argument (and instances inside the method) as well, otherwise it implies the `iterate` methods only work with the `DP5Solver`.